_api-reference/ingest-apis/processors/convert.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: Convert
+parent: Ingest processors 
+grand_parent: Ingest APIs
+nav_order: 30
+---
+
+# Convert
+
+The `convert` processor converts a field in a document to a different type, for example, a string to an integer or an integer to a string. For an array field, all values in the array are converted. The syntax for the `convert` processor is: 
+
+```json
+{
+    "convert": {
+        "field": "field_name",
+        "type": "type-value"
+    }
+}
+```
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `convert` processor.   
+
+**Parameter** | **Required** | **Description** |
+|-----------|-----------|-----------|
+`field`  | Required  | Name of the field whose value to convert.  |
+`type`  | Required  | The type to convert the field value to. The supported types are `integer`, `long`, `float`, `double`, `string`, `boolean`, `ip`, and `auto`. If the `type` is `boolean`, the value is set to `true` if the field value is a string `"true"` (ignoring case), and to `false` if  the field value is a string `"false"` (ignoring case). If the value is not one of the allowed values, an error will occur.  |
+`description`  | Optional  | Brief description of the processor.  |  
+`target_field`  | Optional  | Name of the field to store the converted value. If not specified, the value will be stored in-place in the `field` field. Default is `field`.  |
+`if`  | Optional  | Conditional expression that determines whether the processor should be deployed.  |
```

Comment:
@heemin32 Will you provide an `if` configuration example? I'm getting an error when I test. Here's an example pipeline I'm working with. Thank you. 
```json
PUT _ingest/pipeline/convert-price
{
  "description": "Pipeline that converts price to floating-point number",
  "processors": [
    {
      "convert": {
        "field": "price",
        "type": "string",
        "target_field": "price_float",
        "ignore_missing": false, 
        "on_failure": [
          {
            "set": {
              "field": "error",
              "value": "Failed to convert price"
            }
          }]
      }
    }
  ]
}
```

Reply:
```
{
  "description": "Pipeline that set price as zero when the price value is below zero",
  "processors": [
    {
      "set": {
        "field": "price",
        "value": "0",
        "if": "ctx.price < 0"
      }
    }
  ]
}
```